src/main/webapp/app/exercises/quiz/participate/quiz-participation.component.ts

```diff
@@ -772,17 +774,27 @@ export class QuizParticipationComponent implements OnInit, OnDestroy {
      * Callback method to be triggered when the user changes any of the answers in the quiz (in sub components based on the question type)
      */
     onSelectionChanged() {
-        this.applySelection();
-        if (this.sendWebsocket) {
-            if (!this.disconnected) {
-                // this.isSaving = true;
-                this.submission.submissionDate = this.serverDateService.now();
-                this.sendWebsocket(this.submission);
-                this.unsavedChanges = false;
-                this.updateSubmissionTime();
-            } else {
-                this.unsavedChanges = true;
-            }
+        this.unsavedChanges = true;
+    }
+
+    triggerSave(resetAutoSaveTimer = true): void {
+        if (resetAutoSaveTimer) {
+            this.autoSaveTimer = 0;
+        }
+        if (this.unsavedChanges && !this.isSubmitting) {
+            this.applySelection();
+            this.submission.submissionDate = this.serverDateService.now();
+            this.quizParticipationService
+                .saveOrSubmitForLiveMode(this.submission, this.quizId, false)
+                .pipe(take(1))
+                .subscribe({
+                    next: () => {
+                        // TODO: Do we need the response for anything here?
+                        this.unsavedChanges = false;
+                        this.updateSubmissionTime();
+                    },
+                    error: (error: HttpErrorResponse) => this.onSaveError(error.message), // TODO: Test this. It's not super clear what onSaveError expects.
```

Comment:
Shouldn't this be handled by the `alert.service.ts` ?

Reply:
Custom error handling is needed here to reset the `unsavedChanged` flag. Not sure about the alert itself